iocore/net/SSLConfig.cc

```diff
@@ -426,20 +427,31 @@ SSLConfigParams::getClientSSL_CTX() const
   return client_ctx;
 }
 
+void
+SSLClientCoordinator::reconfigure()
+{
+  // The SSLConfig must have its configuration loaded before the SNIConfig.
+  // The SSLConfig owns the client cert context storage and the SNIConfig will load
+  // into it.
+  SSLConfig::reconfigure();
+  SNIConfig::reconfigure();
+}
+
 void
 SSLConfig::startup()
 {
-  sslConfigUpdate.reset(new ConfigUpdateHandler<SSLConfig>());
-  sslConfigUpdate->attach("proxy.config.ssl.client.cert.path");
-  sslConfigUpdate->attach("proxy.config.ssl.client.cert.filename");
-  sslConfigUpdate->attach("proxy.config.ssl.client.private_key.path");
-  sslConfigUpdate->attach("proxy.config.ssl.client.private_key.filename");
+  sslClientUpdate.reset(new ConfigUpdateHandler<SSLClientCoordinator>());
```

Comment:
Why don't you have this line in `SSLClientCoordinator::startup()` ?

Reply:
I didn't create a SSLClientCoordinator::startup().  Only introduced the class for the reconfigure case.  We hadn't seem ordering problems on a process start.

Probably to be consistent, we could insert SSLClientCoordinator in the startup path as well.

Reply:
Yeah, but not only for consistency. If `SSLClientCoordinator` is responsible for making sure the two configurations are loaded in the right order, I think it should take all the responsibility include the startup path.